scripts/zkai_larger_native_block_boundary_amortization_budget_gate.py

```
@@ -0,0 +1,765 @@
+    actual_source_paths = [require_str(row.get("path"), "source path") for row in source_artifacts]
+    expect_equal(actual_source_paths, expected_source_paths, "source artifact paths")
+    if expected_source_artifacts is None:
+        expected_source_artifacts = [
+            require_dict(row, "expected source artifact row")
+            for row in require_list(load_sources().get(SOURCE_ARTIFACTS_KEY), "expected source artifacts")
+        ]
+    expect_equal(source_artifacts, expected_source_artifacts, "source artifact descriptor")
+
+    non_claims = require_list(payload.get("non_claims"), "non-claims")
+    if not set(NON_CLAIMS).issubset(set(non_claims)):
+        raise AmortizationBudgetError("payload missing non-claims")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)